repose-aggregator/components/filters/api-validator-filter/src/main/java/org/openrepose/filters/apivalidator/ApiValidatorHandler.java

```diff
@@ -153,9 +146,16 @@ public void doFilter(HttpServletRequest request, HttpServletResponse response, F
                 }
 
                 if (!isValid) {
-                    if (mbcsInvalidRequests != null) {
+                    if (Optional.ofNullable(metricsService).isPresent()) {
```

Comment:
Do we actually use Spring Boot to support any part of Repose? For the auto-wiring, we use Spring's IOC API directly (See: `CoreSpringProvider`). I'm not sure if Spring Boot give you another way to auto-wire things, but I don't think we're using it anyway.

To get back on topic, I think you're about Spring wiring in `null`, but the metrics service is a core service. Which is to say, it should never be null. While it isn't strictly necessary for Repose to run, it might be better to treat it like it is. Otherwise, a user might not get the metrics they are expecting. That is, drop the `null` check and let the exception bubble up if we are in a case we should never be in. We don't check that other service beans are non-null, do we?